sensor/common/imagecacheutils/common.go

```diff
@@ -15,3 +15,11 @@ func GetImageCacheKey(provider CacheKeyProvider) string {
 	}
 	return provider.GetName().GetFullName()
 }
+
+// CompareImageCacheKey given two CacheKeyProvider, compares if they're equal
+func CompareImageCacheKey(a CacheKeyProvider, b CacheKeyProvider) bool {
+	if a.GetId() != "" && b.GetId() != "" {
+		return a.GetId() == b.GetId()
+	}
+	return a.GetName().GetFullName() == b.GetName().GetFullName()
```

Comment:
Since `a.GetName()` is a proto message, you can compare them with `proto` package, like:
```suggestion
	return proto.Equal(a.GetName(), b.GetName())
```

Reply:
I kind of want to keep it like that. I’ve seen in other parts of the code that we only compare the ID and, if this one is missing, the FullName, `proto.Equal` will compare all the contents of the `ImageName` but in some cases some other fields might not be set. For example; when we received the `InvalidateImageCache` the content of that request does not have all the fields in the `ImageName` only `Id` and `ImageFullName`.